### ISSUE TITLE
Rare crash in -[FMMixingAudioPlayer preparingAssetDidFail]

### ISSUE THREAD

**Issue by averello** [feedfm/iOS-SDK#4]
Hello,

Using version 4.0.10 from Cocoapods,
we had a few crashes in `-[FMMixingAudioPlayer preparingAssetDidFail]`.
They are not occurring very often to be something concerning or urgent.

I thought you would want to investigate a little further.

This is what Fabric reports:
```
CRASH_INFO_ENTRY_0
Assertion failed: (_preparingFMAsset != nil), function -[FMMixingAudioPlayer preparingAssetDidPrepare], file /Users/eml/git/iOS-SDK-Core/FeedMediaCore/FeedMediaCore/Internal/FMMixingAudioPlayer.m, line 229.
```

and the backtrace:
```
Crashed: com.apple.main-thread
0  libsystem_kernel.dylib         0x18c5c1014 __pthread_kill + 8
1  libsystem_pthread.dylib        0x18c68b264 pthread_kill + 112
2  libsystem_c.dylib              0x18c5359c4 abort + 140
3  libsystem_c.dylib              0x18c50a53c basename_r + 314
4  FeedMedia                      0x101532328 -[FMMixingAudioPlayer preparingAssetDidFail] (FMMixingAudioPlayer.m:250)
5  FeedMedia                      0x1015321c4 __43-[FMMixingAudioPlayer possiblyPrepareAsset]_block_invoke (FMMixingAudioPlayer.m:216)
6  FeedMedia                      0x10152b244 __28-[FMAsset completeWithItem:]_block_invoke (FMAsset.m:94)
7  libdispatch.dylib              0x18c47e9e0 _dispatch_call_block_and_release + 24
8  libdispatch.dylib              0x18c47e9a0 _dispatch_client_callout + 16
9  libdispatch.dylib              0x18c4835e8 _dispatch_main_queue_callback_4CF + 996
10 CoreFoundation                 0x18d5750c8 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 12
11 CoreFoundation                 0x18d572ce4 __CFRunLoopRun + 1572
12 CoreFoundation                 0x18d4a2da4 CFRunLoopRunSpecific + 424
13 GraphicsServices               0x18ef0d074 GSEventRunModal + 100
14 UIKit                          0x19375dc9c UIApplicationMain + 208
15 BlackJack                      0x1001af100 main (main.m:15)
16 libdyld.dylib                  0x18c4b159c start + 4
```

Thanks,
Georges

**Comment by eml** [comment]
Thank you for this! I had thought we fixed these with our latest change to 4.0.10, but apparently something is still sneaking in. Hopefully we'll have an update this week.

**Comment by averello** [comment]
Hi @eml,

we haven't seen this crash in a very long time. Maybe it is fixed and it can be closed.
